yellowbrick/features/base.py

```diff
@@ -309,3 +312,130 @@ def _determine_target_color_type(self, y):
+## Projection Visualizers
+##########################################################################
+
+class ProjectionVisualizer(DataVisualizer):
+    
+    def __init__(self, transformer, ax=None, features=None, classes=None, color=None,
+             colormap=None, target_type="auto", proj_dim=2, alpha=0.75, **kwargs):
+        super(ProjectionVisualizer, self).__init__(ax=ax, features=features, 
+                                                     classes=classes, color=color,
+                                                     colormap=colormap, 
+                                                     target_type=target_type, **kwargs)
+        self.transformer=transformer
+        if proj_dim not in [2, 3]:
+            raise YellowbrickValueError("Projection dimensions must be either 2 or 3")
+        self.proj_dim = proj_dim
+        self.alpha = alpha
+        self._cax = None
+
+    @property
+    def cax(self):
+        """
+        The axes of the colorbar, right of the scatterplot.
+        """
+        if self._cax is None:
+            raise AttributeError(
+                "this visualizer does not have a histogram for the X axis"
+            )
+        return self._cax       
+        
+    def _layout_colorbar(self):
+        """
+        Creates the layout for colorbar. The colorbar is added to the right of the 
+        scatterplot 
+        """
+        # Ensure matplotlib version compatibility
+        if make_axes_locatable is None:
+            raise YellowbrickValueError((
+                "heatmap requires matplotlib 2.0.2 or greater "
+                "please upgrade matplotlib or set heatmap=False on the visualizer"
+            ))
+
+        # Create the new axes for the colorbar
+        divider = make_axes_locatable(self.ax)
+        self._cax = divider.append_axes("right", size="5%", pad=0.3)
+        
+    def fit(self, X, y=None, **kwargs):
+        
+        self.fit_transform(X, y, **kwargs)
+        return self
+        
+    def fit_transform(self, X, y=None, **kwargs):
+
+        # Transformer will be  defined either in the subclasses or explicitly
+        self.transformed_features_ = self.transformer.fit_transform(X)
+        super(ProjectionVisualizer, self).fit(X, y, **kwargs)
+        return self.transformed_features_    
+        
+    def transform(self, X, y=None, **kwargs):
+        try:
+            return self.transformer.transform(X)
+        except AttributeError as e:
+            raise AttributeError(str(e) + " try using fit_transform instead.")
+
+    # Not sure about the function signature      
+    def draw(self, X=None, y=None):
+        X = self.transformed_features_
+        scatter_kwargs = self._determine_scatter_kwargs(y);
+        
+        if self.proj_dim == 2:
+            self._scatter = self.ax.scatter(X[:,0], X[:,1], **scatter_kwargs)
+
+        if self.proj_dim == 3:
+            self.fig = plt.figure()
+            self.ax = self.fig.add_subplot(111, projection='3d')
```

Comment:
What about the case where the user has passed in an `Axes` object? Let's refactor this to override the `Visualizer.ax` property and do appropriate checks as needed.

Reply:
Hey @bbengfort, 
Can you elaborate on what is needed here? I understand that when user passes `Axes` we get axes lines on the plot. To deal with that we just need to clear the figure and redefine the `Axes`. Is that what you mean here?

Reply:
So a couple of things here, maybe I'm misunderstanding - but there is a difference between `Axes` and `axis spines` (see [anatomy of a matplotlib plot](https://www.scikit-yb.org/en/latest/matplotlib.html#figures-and-axes)). The `Axes` has nothing to do with lines on the plot, it is a part of the figure that is drawn upon. 

Consider the following use case:

```python
fig, axes = plt.subplots(ncols=2, projection='3d', figsize=(18,6))

Manifold(ax=axes[0], projection='3d').fit_transform(X, y)
PCA(ax=axes[1], projection='3d').fit_transform(X, y)
```

In this code, the figure is composed of two subplots represented by two Axes objects. The user is asking the Manifold visualization to be drawn on the left axes and the PCA visualization to be drawn on the right axes. 

Because of this we can't just create a figure or an axes on our own - we have to respect what the user passes in if they do, and only create the axes if nothing has been supplied. 

Does that help clear up the confusion?

Reply:
Ok! Great! But what if the user passes non-3d axes? We can't draw 3d plots on that! Also, I could not find methods to convert 2d to 3d axes. Will we raise an error in this case?

Reply:
This is a very good point - we should see what happens! If matplotlib raises a sensible error, then we'll keep that error, or wrap it in a YellowbrickError. Otherwise matplotlib might just plot it in 2D, I'm not sure - so let's find out!